mira/dkg/client.py

```diff
@@ -249,30 +251,25 @@ def get_node_counter(self) -> Counter:
         return Counter(counter_data)
 
     def search(self, query: str, limit: int = 25) -> List[Entity]:
-        """Search nodes for a given name substring."""
+        """Search nodes for a given name or synonym substring."""
         query_lower = query.lower().replace("-", "").replace("_", "")
         cypher = dedent(
             f"""\
-            MATCH (n) 
-            WHERE 
+            MATCH (n)
+            WHERE
                 replace(replace(toLower(n.name), '-', ''), '_', '') CONTAINS '{query_lower}'
                 OR any(
                     synonym IN n.synonyms 
                     WHERE replace(replace(toLower(synonym), '-', ''), '_', '') CONTAINS '{query_lower}'
                 )
-            RETURN n 
-            LIMIT {limit}
+            RETURN n
+            LIMIT {max(limit, 50)}
```

Comment:
The idea here is to make the match larger than the returned list because matching is unranked and sometimes misses entries that would otherwise be highly ranked.